AjobK/Athena
# 259 homepage infinite scrolling v2

closes #259 

<br />

 **What is it supposed to do**  
Fixes the issue from the previous infinite scrolling PR ( #275 ) where the previous post blocks wouldn't fill when loading more posts.

Current state: 
![image](https://user-images.githubusercontent.com/14930277/142760059-0d31d647-22c9-41a8-83e6-b3915e3b8047.png)

New state:
![image](https://user-images.githubusercontent.com/14930277/142760085-6fd4108f-41b8-429f-8711-02b87f50d9e2.png)


<br />

 **How can we test your branch**  
Try adding/removing posts and scroll down (home page) to see if post blocks get filled properly, and if there is no unwanted behavior like posts disappearing.

<br />

 **Additional notes**  
Since posts get fetched in pages of 6 posts, the pages won't end 'cleanly' on a fetch when one page doesn't give 6 posts (they will only get filled at the next fetch, when the user scrolls down):
![image](https://user-images.githubusercontent.com/14930277/142759746-17000f34-8a5d-4fba-8255-27dbd733d765.png)
If this is undesired I could fix this by simply fetching again when the fetch results don't contain 6 posts, so the user doesn't get to see it as much. I could also increase the page length to get the same result (which will most likely be done in the future anyway as fetching once every 6 posts can be expensive). Note that this only occurs once the end of the homepage is reached, which I assume won't happen often in production.

<br />

**1. General checklist**

<br />

**2. If relevant, front-end checklist**

<br />

**3. If relevant, back-end checklist**

<br />

**4. If relevant, test these browsers**
